### PR TITLE
Added logic for collapsing leaves using SAH

### PR DESCRIPTION
**Pull request by elroychan**
First part of collapsing leaf nodes based on lower SAH.

Second part is the work on ComputeAABBs.hlsli and TraverseFunction.hlsli etc, to construct the AABBNodes properly and Traverse through triangle array leaf nodes.

### CODE REVIEW

File: Libraries/D3D12RaytracingFallback/src/ComputeAABBs.hlsli [hunk 1]
```diff
@@ -115,7 +115,7 @@ void main(uint3 DTid : SV_DispatchThreadID)
 
             boxData = GetBoxFromChildBoxes(leftBox, leftNodeIndex, rightBox, rightNodeIndex, outputFlag);
 
-#if COMBINE_LEAF_NODES
+#if COMBINE_LEAF_NODES_1
```
Comment (pkristof) [inline Libraries/D3D12RaytracingFallback/src/ComputeAABBs.hlsli]: Is "_1" intended?

File: Libraries/D3D12RaytracingFallback/src/RayTracingHelper.hlsli [hunk 2]
```diff
@@ -25,6 +25,9 @@ static const int IsLeafFlag = 0x80000000;
 static const int IsProceduralGeometryFlag = 0x40000000;
 static const int LeafFlags = IsLeafFlag | IsProceduralGeometryFlag;
 
+#define COMBINE_LEAF_NODES 1
+static const int IsCollapseChildren = 0x80000000; // for extracting HierarchyNode::bCollapseChildren
```
Comment (pkristof) [inline Libraries/D3D12RaytracingFallback/src/RayTracingHelper.hlsli]: If this is intended to be exclusive to some other value, it should be all defined closely together.

File: Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl [hunk 3]
```diff
@@ -17,7 +17,8 @@
 #include "TreeletReorderBindings.h"
 #include "RayTracingHelper.hlsli"
 
-static const float CostOfRayBoxIntersection = 1.0;
+static const float CostOfRayBoxIntersection = 1.2;
```
Comment (pkristof) [inline Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl]: Please document how these values were chosen, even if it's gut feel.

File: Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl [hunk 4]
```diff
@@ -131,8 +134,10 @@ void FindOptimalPartitions(in uint threadId)
     GroupMemoryBarrierWithGroupSync();
 
     [unroll]
```
Comment (pkristof) [inline Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl]: The unroll should be right above the loop, after a comment

File: Libraries/D3D12RaytracingFallback/src/TreeletReorderBindings.h [hunk 5]
```diff
@@ -74,15 +74,16 @@ static const uint BitPermutations[FullTreeletSize - 2][35] = {
 
 uint GetBitPermutation(uint numBitsSet, uint n)
 {
-    if (numBitsSet == 0)
+    // GetBitPermutation is always called with numBitsSet == [2, FullTreeletSize]
+    /*if (numBitsSet == 0)
```
Comment (pkristof) [inline Libraries/D3D12RaytracingFallback/src/TreeletReorderBindings.h]: Nuke this?

File: Libraries/D3D12RaytracingFallback/src/ComputeAABBs.hlsli (hunk 1 above)
Comment (elroychan) [reply Libraries/D3D12RaytracingFallback/src/ComputeAABBs.hlsli]: Yes, I want this to be 'commented' out. This is part of the 2nd half of the work to collapse leaf nodes, that is not done yet.

**Pull request merged**